### ISSUE TITLE
Wrong hash for leaflet.css

### ISSUE THREAD

**Issue by Swannbm** [RealOrangeOne/django-sri#66]
Hello,

I encoutered a bug with the following file: https://sparte.beta.gouv.fr/static/carto/css/leaflet.css

Provided hash with static_sri : y4gDEU/pxJrmohmEZFiWzZiKDba4qNsDmUTkUJjIN5U=
Expected hash : BPfK9M5v34c2XP6p0cxVz1mUQLst0gTLk0mlc7kuodA= (computed by chrome)

I am unsure on how this is possible, I am looking into it but with no result yet.

Swann

**Comment by Swannbm** [comment]
I might have a lead. It seems that reading a file then convert it in byte is not the same as reading it in byte mode from scratch.

```
with open(path, "rb") as f:
        content = f.read()
```
is not the same as :
```
with open(path, "r") as f:
        content = f.read()
content.decode()
```

I don't know why...

I will make a pull request.

**Comment by Swannbm** [comment]
Current version (0.3.0) has the bug, but it is fixed on master. When do you think you will be able to publish 0.4 on pypi ?

**Comment by RealOrangeOne** [comment]
How strange, I suspect it has something to do with default file encodings? Either way, the latter example is always unnecessary if the file can be read as bytes anyway (it's much faster that way).

I've released 0.4.0 now, which does better deal with opening files correctly, but also in chunks which should improve performance.

Try giving that package a go and let me know if it's working.
